### ISSUE TITLE
HTTP/3: Consider adding libmsquic to mcr.microsoft.com/dotnet/aspnet

### ISSUE THREAD

**Issue by JamesNK** [dotnet/aspnetcore#37311]
Today someone must update their dockerfile to add libmsquic if they want HTTP/3.

We should consider adding libmsquic to mcr.microsoft.com/dotnet/aspnet by default. This issue is mostly for tracking. We should talk to get dockerfile maintainers about whether this is a good idea and how to make it happen.

**Comment by adityamandaleeka** [comment]
@wfurt @JamesNK Has there been any further discussion about this?

**Comment by JamesNK** [comment]
No

**Comment by wfurt** [comment]
no. also this would not be the repro to do the work. since we plan to mark it still preview in 7.0 I don't think dragging extra dependency (and maintenance) make sense. cc: @karelz

**Comment by rchoffardet** [comment]
Hey 👋 
I just did a quick experiment on my side with sending HTTP/3 requests. I was quite surprised that even though .NET 10 seems to support HTTP/3 out of the box, requests are throwing errors by default when forcing HTTP/3 (`RequestVersionExact`). Even more surprised that there is currently no base image that ships with libmsquic and libnuma (its dependency).

Is microsoft planning to ship an image with libmsquic in the short terms? Or should we consider maintaining our own base image? We were reyling on `dotnet publish` to build our artifacts, so far. This was very convenient, so far 🤗 !

**Comment by wfurt** [comment]
we may revisit this when https://github.com/microsoft/msquic/issues/5976 lands. It brings 10x size reduction and removes big chunk of code that needs to be serviced (quiclts)
